### ISSUE TITLE
System end up in a bad state when adding a class in a trait composition instead of a trait

### ISSUE THREAD

**Issue by jecisc** [guillep/pharo-core#6186]
**Describe the bug**
While writing a test I misstyped and set up a Class as a trait composition of another class. 

After that, the system react pretty badly. I cannot remove the trait composition. I cannot remove the class. I cannot remove the package...

**To Reproduce**

```Smalltalk
	| class trait |
	class := Object
		subclass: #ChanelMethodOnlyCallingSuperFake
		instanceVariableNames: ''
		classVariableNames: ''
		package: self class package name.
	trait := Object
		subclass: #TChanelMethodOnlyCallingSuperFake
		instanceVariableNames: ''
		classVariableNames: ''
		package: self class package name..

	class setTraitComposition: trait.
```

**Expected behavior**
I would prefer to get an explicit error during the compilation saying that it is not possible to use a class as a trait composition.

**Version information:**
 - Version: Pharo 8

**Comment by Ducasse** [comment]
Thanks. I cannot see in your example that you added a class as a composition. 
```
Object subclass: #2643
    uses: Point
...
```
Is it it?

**Comment by jecisc** [comment]
`class setTraitComposition: trait`

**Comment by Ducasse** [comment]
Yes I read  it but I do not get it. 
So we will see.

**Comment by Ducasse** [comment]
Could you have a reproducible case? Because else we have to try and see.

**Comment by Ducasse** [comment]
Ah I see trait is a class!

**Comment by tesonep** [comment]
I will fix it. 
Do we need to add a validation also if you put your cat in the microwave? :)

**Comment by tesonep** [comment]
Fixed in #6214

**Comment by MarcusDenker** [comment]
this was fixed
